lib/core/sdam/topology_description.js

```diff
@@ -132,6 +132,8 @@ class TopologyDescription {
     let maxElectionId = this.maxElectionId;
     let commonWireVersion = this.commonWireVersion;
 
+    serverDescription = serverDescription.resetIfWrongSetName(setName);
```

Comment:
I have a medium preference for bringing the logic inside of `resetIfWrongSetName` back into this method, and not hiding it within the implementation details of the `ServerDescription` itself. Perhaps if we were dealing with a [rule of threes](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) situation where this method would be used often in disparate areas of code it would make sense to make this refactor, but as it stands it requires the reader to jump between files to understand the logical flow of what's happening in this particularly complex piece of machinery.